urllib3/_collections.py

```diff
@@ -227,20 +227,20 @@ def add(self, key, val):
                 # Need to convert the tuple to list for further extension
                 _dict_setitem(self, key_lower, [vals[0], vals[1], val])
 
-    def extend(*args, **kwargs):
+    def extend(self, *args, **kwargs):
         """Generic import function for any type of header-like object.
         Adapted version of MutableMapping.update in order to insert items
         with self.add instead of self.__setitem__
         """
-        if len(args) > 2:
-            raise TypeError("update() takes at most 2 positional "
+        if len(args) > 1:
+            raise TypeError("extend() takes at most 1 positional "
                             "arguments ({} given)".format(len(args)))
-        elif not args:
-            raise TypeError("update() takes at least 1 argument (0 given)")
-        self = args[0]
-        other = args[1] if len(args) >= 2 else ()
+        other = args[0] if len(args) >= 1 else ()
         
-        if isinstance(other, Mapping):
+        if isinstance(other, HTTPHeaderDict):
```

Comment:
I'd agree to prefer duck typing, but on which methods distinct from a Mapping do you want to rely? I'm not sure if iteritems is the definite choice here, though it would work for now. Still, would be nice, if the behaviour of all the iterations would have a better defined behaviour, i.e. extracting multiheaders on iteration or not. keys and items handle this different right now, there are reasons for both and it's still not discussed.